tests/test_vqe.py

```diff
@@ -139,39 +228,23 @@ def test_circuits_valid_init(self, ansatz, observables, mock_device):
         assert all(c.device == mock_device for c in circuits)
         assert all(hasattr(c, "jacobian") for c in circuits)
 
-    @pytest.mark.parametrize("ansatz, params", [
-        (EMPTY_ANSATZ, EMPTY_PARAMS),
-        (CUSTOM_FIXED_ANSATZ, EMPTY_PARAMS),
-        (CUSTOM_VAR_ANSATZ, VAR_PARAMS),
-        (strong_ent_layer, LAYER_PARAMS),
-        # FIXME uncomment when https://github.com/XanaduAI/pennylane/issues/365 is addressed
-        #(amp_embed, EMBED_PARAMS),
-        #(amp_embed_and_strong_ent_layer, (EMBED_PARAMS, LAYER_PARAMS)),
-    ])
+    @pytest.mark.parametrize("ansatz, params", CIRCUITS)
     @pytest.mark.parametrize("observables", OBSERVABLES)
-    def test_circuits_evaluate(self, ansatz, observables, params, mock_device):
+    def test_circuits_evaluate(self, ansatz, observables, params, mock_device, seed):
         """Tests that the circuits returned by ``vqe.circuits`` evaluate properly"""
         mock_device.num_wires = 3
         circuits = qml.vqe.circuits(ansatz, observables, device=mock_device)
-        for idx, c in enumerate(circuits):
-            val = c(params)
-            assert type(val) == float
```

Comment:
I did this one as a for loop in case one of the earlier test cases failed, we wouldn't need to evaluate them all

Reply:
:+1:

I changed to list comprehension just so it would be easier to tell which iteration of the loop failed.

Reply:
how does the list comprehension make it easier?

Reply:
It moves the asserts from outside of the loop. This way you evaluate all the booleans first, and then assert all are True. If one is not True, you can pinpoint exactly which circuit is failing.